File: backend/ReceiptWarranty.Api/Models/Receipt.cs
using MongoDB.Bson.Serialization.Attributes;

namespace ReceiptWarranty.Api.Models
{
    public class Receipt
    {
        [BsonId]
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public string Title { get; set; } = string.Empty;
        public string Store { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public string Currency { get; set; } = "SEK";
        public string Category { get; set; } = string.Empty;
        public DateTime PurschaseDate { get; set; }
        public int WarrantyMonths { get; set; }
        public DateTime WarrantyEndDate { get; set; }
        public string? Notes { get; set; }
        public string? ImageURL { get; set; }
    }
}

File: frontend/vaultify-frontend/src/types/receipt.ts
export type currency = "SEK" | "EUR" | "USD";

export interface ReadReceiptDto {
    id: string;
    title: string;
    store: string;
    price: number;
    currency: currency;
    category: string;
    purchaseDate: string;
    warrantyMonths: number;
    warrantyEndDate: string | null;
    notes: string | null;
    imageURL: string | null;
}

export interface CreateReceiptDto {
    title: string;
    store: string;
    price: number;
    currency: currency;
    category: string;
    purchaseDate: string;
    warrantyMonths: number;
    notes?: string | null;
    imageURL: string | null;
}

export type UpdateReceiptDto = CreateReceiptDto;